Bugfix + dockerization

florianhofhammer opened the pull request:
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some documentation

**Description**

Hi,

I tried to use r2wars with RISC-V bots and figured out that it was not working because the `initstate` array in [csharp/clsEngine.cs](https://github.com/radareorg/r2wars/blob/86854d490db1a5c261089a200d692fd1b88c04ce/csharp/clsEngine.cs#L360) is indexed based on the architectures enum but was too small to actually have enough space for all the different architectures.
One of my commits should fix this issue.

While in the process, I also rewrote the Dockerfile and added a compose.yaml for docker compose which made it a lot easier and faster for me to rebuild (both r2wars and radare2).

Please let me know if you think those patches are useful. Please also let me know whether you'd like me to split the commits up into distinct PRs and/or add a few lines to the readme.

Thanks!

trufae commented:
Amazing! Happy to see some movement here!

The patches look good to me. Only two small comments from me:

- is the timezone setup really necessary? Not everyone lives in zurich
- Can you keep the makefile?

florianhofhammer commented:
Hi, thanks for the feedback!

Timezone setup unfortunately is necessary for package installation. Even with the `--no-install-recommends` flag to `apt`, `tzdata` is installed as a dependency and interactively enforces timezone setup at install time. Setting up a timezone already beforehand gets rid of this issue.  
I've set the timezone as a build argument so that it can easily be overridden on the command line when building the Docker image. Nevertheless, I've changed it to UTC to be a bit more generic :)

I've added a Makefile for building, starting, stopping the container. Let me know what you think!

trufae commented:
All good now! :) thanks!
